src/Tasks/Microsoft.NET.Build.Tasks/ResolveReadyToRunCompilers.cs

```diff
@@ -181,42 +181,19 @@ private bool GetCrossgen2TargetOS(out string targetOS)
 
             // Determine targetOS based on target rid.
             // Use the runtime graph to support non-portable target rids.
+            // Use the full target rid instead of just the target OS as the runtime graph
+            // may only have the full target rid and not an OS-only rid for non-portable target rids
+            // added by our source-build partners.
             var runtimeGraph = new RuntimeGraphCache(this).GetRuntimeGraph(RuntimeGraphPath);
             string portablePlatform = NuGetUtils.GetBestMatchingRid(
                     runtimeGraph,
-                    _targetPlatform,
-                    new[] { "linux", "linux-musl", "osx", "win", "freebsd", "illumos" },
```

Comment:
A bit ootl here, is Linux-Musl now bundled together with Linux? Or why are we removing it here?

Reply:
The `linux-musl` RID has always had the `linux` RID as a parent. It's not 100% correct, but we can't really change it at this point due to back-compat.

Reply:
Basically, we could have done this all along (just use `linux` for both `linux` and `linux-musl` here), we just didn't.